### ISSUE TITLE
Compliance/Penalty - Invoice

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-compliance#73]
#### Description:

**Note**: Similar work completed as part of #27 & #139

An operator can download a PDF invoice of their penalty.

- [Design in Figma](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?page-id=14307%3A1663&node-id=27273-1200&node-type=frame&viewport=1759%2C-4811%2C0.18&t=P12lPObqdo6WtkoK-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=20107%3A1497&show-proto-sidebar=1)

#### Acceptance Criteria:

As an operator, given that I have a penalty, when I open the Review Penalty Summary screen, then I can click a button to download a penalty invoice in PDF format.

#### Development Checklist:

- [ ] See example code done in:
   - [x] `bciers/apps/compliance/src/app/components/compliance-summary/manage-obligation/review-compliance-summary/ComplianceSummaryReviewComponent.tsx\ComplianceStepButtons`
   - [x] `bciers/apps/compliance/src/app/components/compliance-summary/manage-obligation/review-compliance-summary/ComplianceSummaryReviewComponent.tsx\handleGenerateInvoice`
   - [x] `bciers/apps/compliance/src/app/api/invoice/[id]/route.ts`
   - [x] `bc_obps/compliance/api/_compliance_report_versions/_compliance_report_version_id/invoice.py`
   - [x] `bc_obps/compliance/service/compliance_invoice_service.py` 
   - [x] `bc_obps/compliance/templates/invoice.html`
   - [x] `bc_obps/compliance/tests/api/_compliance_report_versions/_compliance_report_version_id/test_invoice.py`
   - [x] `bc_obps/compliance/tests/service/test_compliance_invoice_service.py`
   - [x] `bc_obps/compliance/tests/service/test_invoice_template.py`
- [ ] Meets the DOD

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By/Blocking
Blocked by #74 and #89 & #191

**Comment by NicoleGovvy** [comment]
Need to confirm content with the business area, assume it will be similar to the compliance obligation invoice.

**Comment by nanyangpro** [comment]
Link to design is updated in the ticket description. Moving back to Backlog for future refinement and implementation.

**Comment by nanyangpro** [comment]
Moving back to Backlog with link to design updated in the ticket description and "Needs Wireframe" label removed (cc: @patrickisaac)

**Comment by patrickisaac** [comment]
Moved ticket to top of `Backlog` in the No Sprint section
cc: @vesselak

**Comment by patrickisaac** [comment]
Refinement Notes:
- @vesselak to add a comment on this ticket
- Will need DC

**Comment by vesselak** [comment]
Post refinement notes:
Moved the reference to #27 and #139 to the top and thank you for adding example of DC

**Comment by BCerki** [comment]
Testing instructions:
- follow the directions in `cas-registration/docs/compliance/generating_penalties.md` to generate some mock penalty data (POs will need a dev to do this for them)
- restart the FE server (I updated route.ts) and navigate to the compliance summary page (e.g. http://localhost:3000/compliance/compliance-summaries/2/manage-obligation-review-summary) and confirm generating a compliance invoice still works (I did some refactoring)
- change the url to replace `manage-obligation-review-summary` with `review-penalty-summary` (if you get an error when you arrive at the page, it most likely means that there's something wrong with the mock data setup)
- click the generate penalty invoice and confirm it matches wireframe (note that operation address shouldn't be on the invoice--it's currently in the design but will be removed shortly)

UPDATE: to test that fee_date has been added, make sure `Invoice Date` appears in the invoice (it will be whatever day you ran the penalty calculation service and created the invoice) and fee_date appears in metabase on the compliance penalty model

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EckQ6-ceNxtGs2ZHNBGKoMwB797Nh8QTeClbWDr4HziTkQ?e=OcPjHw&nav=eyJoIjoiMTE5MTA5MjIwNyJ9).

I can verify that:
- [x] As an operator, given that I have a penalty, when I open the Review Penalty Summary screen, then I can click a button to download a penalty invoice in PDF format.
- [x] The penalty invoice has the following dates:
   - [x] `Invoice Date` - the date the penalty invoice was created in eLicensing
   - [x] `Invoice Due Date` - 30 days after the received date specified in eLicensing
   - [x] `Invoice Printed Date` - the date industry user clicks `Generate Penalty Invoice`

Just a note as well that the operation name displayed in the invoice seems to be derived from administration, created #293 for the cleanup. Moving to Pending PO for review of test. Thank you!

**Comment by vesselak** [comment]
Reviewed testing documentation and accepting ticket
